### DISCUSSION TITLE
Understanding Gosling vs HiGlass functionality

### DISCUSSION THREAD

**Discussion by mbiokyle29** [gosling-lang/gosling.js#908] [Q&A]
Hello,

Hopefully this is the correct place for this sort of question, but I am mostly trying to decipher what the functionality limitations are between Gosling.js and HiGlass. I am hoping to build a set of Gosling visualizations within an existing React application. The things I am hoping to visualize are:
- DNA sequence (i.e. reference sequence, in fasta format)
- Gene Annotations (currently in bed format, but can adapt to another format)
- BAM coverage depth and read pileup
- Variant calls (in VCF format)

After exploring whats functionally possible with Gosling, I ran into multiple notes in the docs about how something is "only possible with a HiGlass server". For example:
<img width="947" alt="Screen Shot 2023-05-23 at 3 00 25 PM" src="https://github.com/gosling-lang/gosling.js/assets/3450485/d49258fe-7f54-4866-8efc-3dfea3909601">

There is also discussion here that references fasta support for Gosling: https://github.com/gosling-lang/gosling.js/issues/184

However, if I jump over to HiGlass, I find the following example: https://higlass.io/app/?config=TA_BP0R5R8GfKHkKX7-TzQ

Which is a sequence + BAM viewer, which is 80% of what I need. When I inspect the viewConf it appears that the only thing coming from the actual HiGlass server is chromosome labels. The fasta and bam files are just URLs. I think I am just confused about the different modalities. I can envision a few different cases:
- Things you can show in Gosling.js without a HiGlass server
- Things you can show in Gosling.js with access to a HiGlass server
- Things you cannot ever show in Gosling.js, regardless of having access to a server. You have to use the HiGlass viewer application

In other words, it would be impossible at this time to build the visualization I want in Gosling.js, even if I have a HiGlass server. A fasta track is just not supported in Gosling.js.
